flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/ExecutingTest.java

```diff
@@ -634,17 +628,23 @@ public ExecutingStateBuilder setLastRescale(Instant lastRescale) {
         private Executing build(MockExecutingContext ctx) {
             executionGraph.transitionToRunning();
 
-            return new Executing(
-                    executionGraph,
-                    getExecutionGraphHandler(executionGraph, ctx.getMainThreadExecutor()),
-                    operatorCoordinatorHandler,
-                    log,
-                    ctx,
-                    ClassLoader.getSystemClassLoader(),
-                    new ArrayList<>(),
-                    scalingIntervalMin,
-                    scalingIntervalMax,
-                    lastRescale);
+            try {
+                return new Executing(
+                        executionGraph,
+                        getExecutionGraphHandler(executionGraph, ctx.getMainThreadExecutor()),
+                        operatorCoordinatorHandler,
+                        log,
+                        ctx,
+                        ClassLoader.getSystemClassLoader(),
+                        new ArrayList<>(),
+                        scalingIntervalMin,
+                        scalingIntervalMax,
+                        lastRescale);
+            } finally {
+                Preconditions.checkState(
```

Comment:
This change makes sense. We don't have any tests that cover this for the issue that's resolved in this PR, though (reverting the change in `Executing` won't cause any failures). We might want to add another test like that one:
```java
    @Test
    public void
            testNotifyNewResourcesAvailableWithCanScaleUpWithoutForceWhileTransitioningIntoExecutingState()
                    throws Exception {
        try (MockExecutingContext ctx = new MockExecutingContext()) {
            final ExecutingStateBuilder executingStateBuilder =
                    new ExecutingStateBuilder()
                            .setScalingIntervalMin(Duration.ofSeconds(20L))
                            .setScalingIntervalMax(Duration.ofSeconds(30L))
                            .setLastRescale(Instant.now().minus(Duration.ofSeconds(25L)));

            // slots become available while transitioning into Executing state
            ctx.setCanScaleUp(true);

            final Executing exec = executingStateBuilder.build(ctx);

            ctx.setExpectRestarting(
                    restartingArguments ->
                            assertThat(
                                    "Rescaling should be triggered immediately after transitioning into Executing state.",
                                    restartingArguments.getBackoffTime(),
                                    is(Duration.ZERO)));
            exec.onNewResourcesAvailable();
        }
    }
```

Reply:
the regression is reliably covered by `testExecutingChecksForNewResourcesWhenBeingCreated`

Reply:
this is what's going to trigger it

```
-        context.runIfState(this, this::rescaleWhenCooldownPeriodIsOver, Duration.ZERO);
+        rescaleWhenCooldownPeriodIsOver();
```

Reply:
true, I didn't see that one because it's not failing (at least on my machine) when running all `ExecutingTest` test methods. Is this the case for you as well? :thinking:

Reply:
Ok, it seems to be an IDE-specific thing. I can reproduce the failure when calling the test from the command line:
```bash
$ /mvnw -pl flink-runtime test -Dtest=ExecutingTest -Dfast
```